kafka_helpers/consumer.go

```diff
@@ -0,0 +1,50 @@
+package kafka_helpers
+
+import (
+	"context"
+	"fmt"
+	"github.com/segmentio/kafka-go"
+	"internship_project/elasticsearch_helpers"
+	"log"
+	"strings"
+)
+
+type KafkaConsumer struct {
+	Reader   *kafka.Reader
+	EsClient elasticsearch_helpers.ElasticsearchClient
+}
+
+func (consumer *KafkaConsumer) Consume() {
+	fmt.Println("KafkaConsumer is ready to consume new messages.")
+	for {
+
+		m, err := consumer.Reader.FetchMessage(context.Background())
+		if err != nil {
+			log.Printf("Error while fetching message")
+		}
+
+		fmt.Printf("message at offset %d: %s = %s\n", m.Offset, string(m.Key), string(m.Value))
+
+		s := strings.Split(string(m.Value), " ")
```

Comment:
Using a simple text protocol is not all bad, but I recommend using a well-known serialization format to pass commands (e.g. JSON)

Reply:
Fixed in the latest commit.